### ISSUE TITLE
How to make filehandles for Webmin open_ calls lexical in scope

### ISSUE THREAD

**Issue by swelljoe** [webmin/webmin#390]
Hey Jamie,

I'm getting stumped by some of the Webmin file operations when trying to make the file handles lexical.

For example, in apache-lib.pl:

```
    my $CONF;
    &open_readfile($CONF, $file);
    @rv = &parse_config_file($CONF, $lnum, $file);
    close($CONF);
```

This fails; no error, but the result set is empty after. Trying to pass $CONF by reference (\$CONF) also fails (with an error). Leaving it with barewords works, but I have to wrap that up in a `no strict "subs";` thing, which is ugly.

Changing as I've done above seems to have worked for *some* of the Webmin file functions (I got it to work for the various file operations in bind-lib.pl, I think), but not this particular function (and a couple of others, though I don't remember which ones...I think the socket one from ajaxterm was a problem child, too).

Got any ideas on how this can be made to work with a lexical scope file handle?

**Comment by swelljoe** [comment]
Note also that the modern Perl approach to declaring file handles is something like this:

```
open(my $CONF, "<", $file);
```

It might be worth making the do-able in the Webmin functions, as well (and it might already work, if I can sort out how to use lexical file handles, at all, since I don't think parameters are in a separate scope from the function call...same block, same scope, I guess.

**Comment by jcameron** [comment]
Yeah, I'm not sure what's going wrong here. The `open_readfile` function just calls `open` with the same parameter. Maybe `open` is special in that it can set the `$CONF` parameter?
